Sources/SwiftSyntax/SyntaxConvenienceMethods.swift

```diff
@@ -11,24 +11,20 @@
 //===----------------------------------------------------------------------===//
 
 public extension FloatLiteralExprSyntax {
-  var floatingValue: Double {
-    return potentialFloatingValue!
-  }
-
-  fileprivate var potentialFloatingValue: Double? {
+  var floatLiteralValue: Double? {
     let floatingDigitsWithoutUnderscores = floatingDigits.text.filter {
       $0 != "_"
     }
     return Double(floatingDigitsWithoutUnderscores)
   }
-}
 
-public extension IntegerLiteralExprSyntax {
-  var integerValue: Int {
-    return potentialIntegerValue!
+  var isValid: Bool {
+    floatLiteralValue != nil
   }
+}
 
-  fileprivate var potentialIntegerValue: Int? {
+public extension IntegerLiteralExprSyntax {
+  var integerLiteralValue: Int? {
```

Comment:
OK, so here's my one comment on API design. I think users will want to know what value is represented by their integer literal expression for types other than the default `Int` and `Double`. It would be very reasonable to know, what's the `UInt32` value represented by a certain literal, for instance.

Therefore, we can use a (generic) method instead of a property to allow the user to pass a variable indicating the desired type:

```swift
func integerLiteralValue<T>(ofType _: T.Type) -> T
where T: ExpressibleByIntegerLiteral { ... }
```

Users could then write: `literalExpr.integerLiteralValue(ofType: Int32.self)`. Now, it's likely that users will most often want a value of the _default_ literal types, so it's reasonable to add that:

```swift
func integerLiteralValue<T>(ofType _: T.Type = IntegerLiteralType) -> T
where T: ExpressibleByIntegerLiteral { ... }
```

Now, the use site simply differs from what you have currently in this PR by a pair of parentheses. Pretty good. Now, if you were to try to implement this API, you'd find that you actually need to constrain `T` a bit further in order to use the String conversion APIs.

It's tempting to write `where T: ExpressibleByIntegerLiteral, T: LosslessStringConvertible`, but in fact this raises a problem: there's no _semantic_ guarantee that the _string format_ from which a type can be losslessly converted is the same as the _literal format_. But in practice, we might be able to get away with it. However, you'd run into another problem when it comes to integer literals, because if you want to pass the `radix` argument, for which you'd need to constrain to `T: FixedWidthInteger` instead. This is fine, but floating-point types are expressible by _integer_ literals but _not_ fixed-width integer types; it'd be difficult to make it possible to convert from binary or octal integer literals to floating-point types without implementing your own conversions from scratch here. This is certainly not necessary for this PR and likely out of scope for this project entirely, but if we ignore floating-point types and integer literals, you can get 80% of the way just by making a few adjustments to the API design.